pkg/lib/csv/replace_finder.go

```diff
@@ -0,0 +1,70 @@
+package csv
+
+import (
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NewReplaceFinder returns an instance of ReplaceFinder
+func NewReplaceFinder(logger *logrus.Logger, client versioned.Interface) ReplaceFinder {
+	return &replace{
+		logger: logger,
+		client: client,
+	}
+}
+
+// ReplaceFinder is an interface that finds the next or previous
+// ClusterServiceVersion object in the upgrade path for a given CSV.
+type ReplaceFinder interface {
+	IsBeingReplaced(in *v1alpha1.ClusterServiceVersion, csvsInNamespace map[string]*v1alpha1.ClusterServiceVersion) (replacedBy *v1alpha1.ClusterServiceVersion)
+	IsReplacing(in *v1alpha1.ClusterServiceVersion) *v1alpha1.ClusterServiceVersion
+}
+
+type replace struct {
+	logger *logrus.Logger
+	client versioned.Interface
+}
+
+// IsBeingReplaced returns the corresponding ClusterServiceVersion object that
+// is replacing the given CSV specified.
+//
+// If the corresponding ClusterServiceVersion is not found nil is returned.
+func (r *replace) IsBeingReplaced(in *v1alpha1.ClusterServiceVersion, csvsInNamespace map[string]*v1alpha1.ClusterServiceVersion) (replacedBy *v1alpha1.ClusterServiceVersion) {
+	for _, csv := range csvsInNamespace {
+		if csv.IsCopied() {
```

Comment:
I checked the call sites of this function, it just requeues both the given CSV and the one replacing it `old -> next` if `next.Status.Phase ~= v1alpha1.CSVPhaseSucceeded`. Let me know if we don't need to requeue copied CSV here.

https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/operators/olm/operator.go#L1094